federation/plumbing/queue_server.py

```diff
@@ -37,10 +41,21 @@ def callback(name, doc):
             try:
                 next_point = adaptive_object.ask(1)
             except NoRecommendation:
-                queue.put(None)
+                queue_server.queue_item_add(None)
```

Comment:
Don't add `None` to the queue. If this is a case where you just don't know which sample to measure and want to wait for a future recommendation just `pass` and tell the queueserver nothing. But maybe you do have a fallback you can do here when no recommendation is made.